test/test.impersonated.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as nock from 'nock';
+import {describe, it, afterEach} from 'mocha';
+import {Impersonated, JWT} from '../src';
+import {CredentialRequest} from '../src/auth/credentials';
+
+const PEM_PATH = './test/fixtures/private.pem';
+
+nock.disableNetConnect();
+
+const url = 'http://example.com';
+
+function createGTokenMock(body: CredentialRequest) {
+  return nock('https://www.googleapis.com')
```

Comment:
Nit: we're trying to centralize the different token exchange endpoints to 'https://oauth2.googleapis.com/token', even though this is a mock, it might be good for new code to just show that one.

Reply:
We would need to change the underlying `gtoken` library to stop using the `/v4` endpoint.

In practice, I think we'll often be using self-signed JWTs for many of the cases where we would have used gtoken in the past. Regardless, I've created a tracking ticket based on this comment:

https://github.com/googleapis/node-gtoken/issues/388